design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,711 @@
+
+#### Conclusion
+- (2) virt-controller will add the network-status to the VMI, the guest will have no disruption and its VFs mapping will not change.
+- (4) When virt-controller pods are is upgraded, new VMs will be created with the correct SR-IOV interfaces assignment.
+
+### Pros
+1. No need to use Downward API and mount its Volume.
+2. No need to read files and parse their content.
+3. No API changes.
+
+### Cons
+1. Require patching the VMI metadata by virt-controller VMIController [[4]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-controller/watch/vmi.go#L1054)  [[5]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-controller/watch/vmi.go#L509).
+
+2. Won't play nice with live migration.<br/>On migration there will be two sources for the networks-status annotation, the migration source and target pods, the VMI might end up with having the network-status annotation with the wrong annotation details.
+
+### Extending each VMI underlying virt-launcher Pod networks-status annotation
+Following Cons (2), in order to ensure that the correct mapping information is passed down to virt-launcher, it is necessary to record each of the VMI virt-launcher pods networks-status annotation.
+
+1. virt-controller look into each of the VMI underlying virt-launcher pods and read their networks-status annotation.
+For each networks-status annotation add an identifier to the annotation name of the pod it belongs (e.g: pod name, UID, etc..) and copy it to the VMI annotations.
+
+#### Example
+```yaml
+kind: VirtualMachine
+...
+metadata:
+  annotations:
+    kubevirt.io/virt-launcher-networks-status: “{
+      “virt-launcher-abc123”: “...”,
+      “virt-launcher-xyz456”: “...”
+  }”
+...
+```
+
+2. virt-launcher pick the relevant details form the annotation and perform the mapping accordingly. 
+
+#### Pros
+1. The correct information required to perform the mapping is passed down even on migration. 
+ 
+#### Cons
+2. Requires virt-controller to monitor and update the annotation content according to the existence of each pod (migration source and target pods).
+
+## 2.Expose Network Status Annotation via VMI metadata by virt-handler
+As part of the virt-handler VM create / update flow, the VMI spec is processed and virt-launcher is triggered to synchronize with the new VMI spec [[1]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-handler/vm.go#L2639) (i.e: sends [SyncVMI](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-launcher/virtwrap/manager.go#L822) gRPC call to virt-launcher).
```

Comment:
maybe worth mentioning that we will need the extended `kubevirt.io/virt-launcher-networks-status` annotations for this as well

Reply:
Note that it is mentioned at line 681, do you think it should be mentioned anyway?